pkg/virt-handler/node-labeller/node_labeller.go

```diff
@@ -0,0 +1,183 @@
+	labellerNamespace = "node-labeller"
+)
+
+var nodeLabellerVolumePath = "/var/lib/kubevirt-node-labeller"
+
+//NodeLabeller struct holds informations needed to run node-labeller
+type NodeLabeller struct {
+	kvmController     *device_manager.DeviceController
+	configMapInformer cache.SharedIndexInformer
+	clientset         kubecli.KubevirtClient
+	host              string
+	namespace         string
+}
+
+func NewNodeLabeller(kvmController *device_manager.DeviceController, configMapInformer cache.SharedIndexInformer, clientset kubecli.KubevirtClient, host, namespace string) *NodeLabeller {
+	return &NodeLabeller{
+		kvmController:     kvmController,
+		configMapInformer: configMapInformer,
+		clientset:         clientset,
+
+		host:      host,
+		namespace: namespace,
+	}
+}
+
+//Run runs node-labeller
+func (n *NodeLabeller) Run(stop chan struct{}) {
+	logger := log.DefaultLogger()
+	logger.Infof("node-labeller is running")
+
+	n.configMapInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			err := n.run()
+			if err != nil {
+				logger.Infof("something happened during labelling node %s at config map creatinon: %v\n", n.host, err)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			err := n.run()
+			if err != nil {
+				logger.Infof("something happened during labelling node %s at config map deletion: %v\n", n.host, err)
+			}
+		},
+		UpdateFunc: func(old, new interface{}) {
+			err := n.run()
+			if err != nil {
+				logger.Infof("something happened during labelling node %s at config map update: %v\n", n.host, err)
+			}
+		},
+	})
+
+	go n.configMapInformer.Run(stop)
+	cache.WaitForCacheSync(stop, n.configMapInformer.HasSynced)
+}
+
+func (n *NodeLabeller) run() error {
+	logger := log.DefaultLogger()
+
+	if !n.kvmController.NodeHasDevice(device_manager.KVMPath) {
+		return goerror.New(fmt.Sprintf("Node-labeller cannot work without KVM device."))
+	}
+
+	cpuFeatures := make(map[string]bool)
+	cpuModels := make([]string, 0)
+	//parse all informations
+	cpuModels, cpuFeatures, err := n.getCPUInfo()
+	if err != nil {
+		logger.Infof("node-labeller cannot get new labels %s\n", err.Error())
+	}
+	//get node on which virt-handler is running
+	node, err := n.clientset.CoreV1().Nodes().Get(n.host, metav1.GetOptions{})
+	if err != nil {
+		logger.Infof("node-labeller cannot get node %s", n.host)
+	}
+
+	//prepare new labels
+	newLabels := n.prepareLabels(cpuModels, cpuFeatures)
+	//remove old labeller labels
+	n.removeCPULabels(node, n.getNodeLabellerLabels(node))
+	//add new labels
+	n.addNodeLabels(node, newLabels)
+	//patch node with new labels
+	data := []byte(fmt.Sprintf(`[{"op": "replace", "path": "/metadata/labels", "value": {%s}}, {"op": "replace", "path": "/metadata/annotations", "value": {%s}}]`, convertMapToText(node.Labels), convertMapToText(node.Annotations)))
+	_, err = n.clientset.CoreV1().Nodes().Patch(node.Name, types.JSONPatchType, data)
```

Comment:
this error is logged, but not really handled. For example, what guarantees if the Patch fails, that it will be tried again at some point?

That's what a work queue gives us, we have the ability to re-enqueue a key when an error is encountered which will guarantee us that the logic will be retried shortly. 

Follow the pattern for `pkg/virt-controller/watch/drain/disruptionbudget/disruptionbudget.go` that's a pretty simple reconcile loop controller.